russstation/code/modules/mob/living/carbon/human/species_types/diona.dm

```diff
@@ -30,60 +47,47 @@
-/datum/species/diona/on_species_loss(mob/living/carbon/C, datum/species/old_species, pref_load)
-	C.draw_russ_parts(TRUE)  //icon path is reset to default
-	. = ..()
-	C.faction -= "plants"
-	C.faction -= "vines"
-
-/datum/species/diona/spec_life(mob/living/carbon/human/H)
+// Similar to podpeople.dm
+/datum/species/diona/spec_life(mob/living/carbon/human/H, delta_time, times_fired)
 	if(H.stat == DEAD)
 		return
+
 	var/light_amount = 0 //how much light there is in the place, affects receiving nutrition and healing
 	if(isturf(H.loc)) //else, there's considered to be no light
 		var/turf/T = H.loc
-		light_amount = min(1,T.get_lumcount()) - 0.5
-		H.adjust_nutrition(light_amount * 10)
+		light_amount = min(1, T.get_lumcount()) - 0.5
+		// Maybe set to `5 *` like podpeople?
+		H.adjust_nutrition(10 * light_amount * delta_time)
 		if(H.nutrition > NUTRITION_LEVEL_ALMOST_FULL)
 			H.set_nutrition(NUTRITION_LEVEL_ALMOST_FULL)
 		if(light_amount > 0.2) //if there's enough light, heal
-			H.heal_overall_damage(2,1, 0, BODYTYPE_ORGANIC)
+			// OLD HEAL CODE
+			// H.heal_overall_damage(2,1, 0, BODYTYPE_ORGANIC)
+			// New Code | Might need to tweak numbers to match podpeople `0.5 * delta_time, 0.5 * delta_time`
+			H.heal_overall_damage(2 * delta_time, 1 * delta_time, 0, BODYTYPE_ORGANIC)
+			H.adjustToxLoss(-0.5 * delta_time)
+			// I assume we don't need this as we have `TRAIT_NOBREATH`
+			// H.adjustOxyLoss(-0.5 * delta_time)
 
 	if(H.nutrition < NUTRITION_LEVEL_STARVING + 50)
-		H.take_overall_damage(2,0)
+		// Possibly reduce to `1 *` similar to podpeople?
+		H.take_overall_damage(2 * delta_time, 0)
+	..()
 
-/datum/species/diona/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H)
+/datum/species/diona/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H, delta_time, times_fired)
 	if(chem.type == /datum/reagent/toxin/plantbgone)
-		H.adjustToxLoss(3)
-		H.reagents.remove_reagent(chem.type, REAGENTS_METABOLISM)
+		H.adjustToxLoss(3 * REAGENTS_EFFECT_MULTIPLIER * delta_time)
+		H.reagents.remove_reagent(chem.type, REAGENTS_METABOLISM * delta_time)
 		return TRUE
 
-/datum/species/diona/on_hit(obj/projectile/P, mob/living/carbon/human/H)
-	switch(P.type)
-		if(/obj/projectile/energy/floramut)
-			if(prob(15))
-				H.AddComponent(/datum/component/irradiated)
-				H.Paralyze(100)
-				H.visible_message(
-					span_warning("[H] writhes in pain as [H.p_their()] vacuoles boil."),
-					span_userdanger("You writhe in pain as your vacuoles boil!"),
-					span_hear("You hear the crunching of leaves."),
-				)
-				if(prob(80))
-					H.easy_random_mutate(NEGATIVE+MINOR_NEGATIVE)
-				else
-					H.easy_random_mutate(POSITIVE)
-				H.random_mutate_unique_identity()
-				H.random_mutate_unique_features()
-				H.domutcheck()
-			else
-				H.adjustFireLoss(rand(5,15))
-				H.show_message(span_userdanger("The radiation beam singes you!"))
-		if(/obj/projectile/energy/florayield)
-			H.set_nutrition(min(H.nutrition+30, NUTRITION_LEVEL_FULL))
```

Comment:
Why are we removing both of these item interactions wholesale?

Reply:
By having the species `inherent_biotypes` include `MOB_PLANT` this interaction is already handled by TG's code in `code\modules\projectiles\projectile\special\floral.dm`

Would be good to double check everything lines up as y'all want though and if not can look to add this back to ensure it works as y'all want.